### ISSUE TITLE
Base search on redendered text and display and highlight rendered text

### ISSUE THREAD

**Issue by felixhayashi** [danielo515/TW5-contextPlugin#5]
Hi @danielo515,

at the moment, the context search only searches in the markup text of the tiddlers. It would be great if the plugin could be extended that it can be configured to perform the search and highlighting based on the rendered markup (I mean what the user actually sees if he/she is not in edit mode).

-Felix

**Comment by felixhayashi** [comment]
To give an example:

The search should be configurable to show:

    ... bla ...

instead of

    ... //bla// ...

Same goes for transcluded text or list widget outputs: Not the markup or code should be shown or searched, but the rendered text.

Hope it got clearer.

-Felix

**Comment by felixhayashi** [comment]
maybe with the help of `$tw.wiki.renderTiddler()` ?

**Comment by danielo515** [comment]
But that means rendering the whole wiki store to search on it. What are you proposing is to search in the rendered tiddlers or just display the result rendered?

**Comment by felixhayashi** [comment]
Yes, I was proposing to render (= 1. creating the html and then 2. search on the html and then 3. highlight the rendered html just like a browser highlights search results when you press ctrl+f or like google highlights search results). No way this is not possible? The point is, when using your plugin in a website, the user will often see snippets of code as a result of the search...

Maybe you have an idea how to handle this.

**Comment by danielo515** [comment]
The part of showing the rendered result, maybe could be possible, but I'm not sure we can search on rendered tiddler. As you said, this mean creating the html, and since we don't know in which tiddler it is located we have to render all the tiddlers and then search through them. In fact I don't know how the search function of tiddlywiki works on that scenario.

**Comment by felixhayashi** [comment]
> As you said, this mean creating the html, and since we don't know in which tiddler it is located we have to render all the tiddlers and then search through them.

Yes, did you take a look at the `renderTiddler()` function I posted above? It is very easy to go through the tiddlers via `this.wiki.allTitles()` and then use `this.wiki.renderTiddler()` It is also quite fast.

For example go to tiddlywiki.com and try:
var allTiddlers = $tw.wiki.allTitles();

    for(var i = 0; i < allTiddlers.length; i++) {
      console.log(allTiddlers[i]);
      if($tw.wiki.isSystemTiddler) continue;
      $tw.wiki.renderTiddler("text/html", allTiddlers[i]);
    }

    console.log(i);

Maybe you know how to proceed from this point?

>  In fact I don't know how the search function of tiddlywiki works on that scenario.

Would be nice if you could have a look at it. I will also have a look at it once I find the time.

I really hope this could be done somehow.

-Felix
